core/src/test/java/org/mskcc/cbio/oncokb/util/IndicatorUtilsTest.java

```diff
@@ -210,13 +210,37 @@ public void testProcessQuery() throws Exception {
+        query = new Query(null, null, null,  null, "MSI-H", null, "Colorectal Cancer", null, null, null, null);
         indicatorQueryResp = IndicatorUtils.processQuery(query, null, null, null, true);
         assertTrue("The geneExist is not false, but it should be.", indicatorQueryResp.getGeneExist() == false);
         assertEquals("The oncogenicity is not Oncogenic, but it should be.", Oncogenicity.YES.getOncogenic(), indicatorQueryResp.getOncogenic());
         assertEquals("The highest sensitive level is not 1, but it should be.", LevelOfEvidence.LEVEL_1, indicatorQueryResp.getHighestSensitiveLevel());
         assertEquals("The gene summary is not empty, but it should be.", "", indicatorQueryResp.getGeneSummary());
 
+        // Test indicator endpoint supports HGVS
+        query = new Query(null, null, null, null, null, null, "Melanoma", null, null, null, "7:g.140453136A>T");
+        indicatorQueryResp = IndicatorUtils.processQuery(query, null, null, null, true);
+        assertTrue("The geneExist is not true, but it should be.", indicatorQueryResp.getGeneExist() == true);
+        assertEquals("The oncogenicity is not Oncogenic, but it should be.", Oncogenicity.YES.getOncogenic(), indicatorQueryResp.getOncogenic());
+        assertEquals("The highest sensitive level is not 1, but it should be.", LevelOfEvidence.LEVEL_1, indicatorQueryResp.getHighestSensitiveLevel());
+
+        query1 = new Query(null, null, null, null, null, null, "Melanoma", null, null, null, "7:g.140453136A>T");
+        query2 = new Query(null, null, null, "BRAF", "V600E", null, "Melanoma", null, null, null, null);
+        resp1 = IndicatorUtils.processQuery(query1, null, null, null, true);
+        resp2 = IndicatorUtils.processQuery(query2, null, null, null, true);
+        assertTrue("Genes are not the same, but they should.", resp1.getGeneSummary().equals(resp2.getGeneSummary()));
+        assertTrue("Oncogenicities are not the same, but they should.", resp1.getOncogenic().equals(resp2.getOncogenic()));
+        assertTrue("Treatments are not the same, but they should.", resp1.getTreatments().equals(resp2.getTreatments()));
+        assertTrue("Highest sensitive levels are not the same, but they should.", LevelUtils.areSameLevels(resp1.getHighestSensitiveLevel(), resp2.getHighestSensitiveLevel()));
+        assertTrue("Highest resistance levels are not the same, but they should.", LevelUtils.areSameLevels(resp1.getHighestResistanceLevel(), resp2.getHighestResistanceLevel()));
+
+        // Test HGVS has higher priority than gene/variant pair
+        // 7:g.140453136A>T is BRAF V600E
+        query = new Query(null, null, null, "ALK", "R401Q", null, "Melanoma", null, null, null, "7:g.140453136A>T");
+        indicatorQueryResp = IndicatorUtils.processQuery(query, null, null, null, true);
+        assertTrue("The geneExist is not true, but it should be.", indicatorQueryResp.getGeneExist() == true);
```

Comment:
Just to keep the consistency, I would prefer assertEquals("The geneExist...", indicatorQueryResp.getGeneExist(), true ). But it is up to you